wordpress-mobile/WordPress-Android
# Disable notifications on WP app in migration flow on JP app

Disable notifications on WP app while in the migration flow to JP app

References: 
- https://github.com/wordpress-mobile/WordPress-Android/pull/17371
- https://github.com/wordpress-mobile/WordPress-Android/pull/17435
- https://github.com/wordpress-mobile/WordPress-Android/pull/17420
- https://github.com/wordpress-mobile/WordPress-Android/pull/17496

| Notifications screen on JP | Notifications settings on WP |
| - | - |
|![Screenshot_20221123_120615](https://user-images.githubusercontent.com/990349/203451279-322b5374-2a7d-41bf-901f-c1ae22b8608a.png)|![Screenshot_20221123_120635](https://user-images.githubusercontent.com/990349/203451314-db8f93ad-5b1f-4527-b635-c8a348fede3f.png)|


To test:

Test 1:
- Ensure either Jetpack app is installed
- Ensure WordPress app is also on the device
- Ensure you already receive some push notifications

Test 2:
- Build and Run WordPress app or install using QR code below
- Build and Run Jetpack app or install using QR code below

On Jetpack app
- Open the Jetpack app & login
- Go to `Me` → `App Settings` → `Debug Settings`
- Enable the `JetpackMigrationFlowFeatureConfig`
- Enable the `prevent_duplicate_notifs_remote_field` 
- Restart the app
- Tap the `Continue` button on the "Welcome to Jetpack!" screen
   - Note: Temporarily the migration progress is simulated by showing for 2 seconds a processing state.
- **Expect** to see the "Notifications now come from Jetpack" screen
- Tap **Continue** on Notifications screen
- Verify `Intent { act=org.wordpress.android.broadcast.DISABLE_NOTIFICATIONS pkg=org.wordpress.android.prealpha }` is fired in Logcat in AS

On WordPress app
- Verify switch is set to Off on Notification Settings ( Home > Notifications > ⚙️ on the top right hand corner )

Test 3
- Wait till the next day if required or [push a debug notification](https://mc.a8c.com/mobile/push-notifications/debug/)
- Verify push notifications are **not** displayed for WordPress app
- Ensure notifications are displayed for Jetpack app as before.

> 🔔 **Notice:** Once the feature flag is enabled the migration flow will always be presented. The Jetpack app build has to be uninstalled (or "Clear Data" should be done) in order to revert back to the normal functionality. Further development on the data layer will make sure the migration is not attempted again once it's successful. To be partially handled in #17406.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually

3. What automated tests I added (or what prevented me from doing so)
Existing unit tests

PR submission checklist:
